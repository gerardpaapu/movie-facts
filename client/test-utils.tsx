import { afterEach, expect } from 'vitest'
import { cleanup, render } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { RouterProvider, createMemoryRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import { initialiseStore } from './store'
import matchers from '@testing-library/jest-dom/matchers'
import { routes } from './routes'

expect.extend(matchers)

afterEach(cleanup)

export function setup(location = '/') {
  const router = createMemoryRouter(routes, {
    initialEntries: [location],
  })

  const container = render(
    <Provider store={initialiseStore()}>
      <RouterProvider router={router} />
    </Provider>
  )

  const user = userEvent.setup()
  return { user, ...container }
}
