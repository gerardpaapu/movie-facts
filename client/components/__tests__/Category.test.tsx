// @vitest-environment jsdom
import nock from 'nock'
import { describe, it, expect, afterEach } from 'vitest'
import { screen, cleanup } from '@testing-library/react'
import { setup } from '../../test-utils'

import matchers from '@testing-library/jest-dom/matchers'

expect.extend(matchers)

afterEach(cleanup)

describe('<Category />', () => {
  it('Shows information about the category', async () => {
    const scope = nock('http://localhost')
      .get('/api/v1/categories/4?withMovies=true')
      .reply(200, {
        id: 4,
        movies: [
          {
            id: 3,
            release_year: 2010,
            title: 'Black Swan',
          },
          {
            id: 11,
            release_year: 2013,
            title: 'Gravity',
          },
        ],
        name: 'Thriller',
      })

    const { container } = setup('/category/4')

    await screen.findByText(/Black Swan/)
    expect(container).toMatchSnapshot()
    expect(scope.isDone()).toBe(true)
  })
})
