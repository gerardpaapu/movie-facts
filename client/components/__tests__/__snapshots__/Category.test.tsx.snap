// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Category /> Shows information about the category 1`] = `
<div>
  <header
    class="header"
  >
    <h1>
      Movie Facts
    </h1>
  </header>
  <nav>
    <ul>
      <li>
        <a
          href="/movie"
        >
          All movies
        </a>
      </li>
      <li>
        <a
          href="/category"
        >
          All categories
        </a>
      </li>
      <li>
        <a
          href="/search"
        >
          Search
        </a>
      </li>
    </ul>
  </nav>
  <section
    class="main"
  >
    <div>
      <h2>
        Category - 
        Thriller
      </h2>
      <ul>
        <li>
          <a
            href="/movie/3"
          >
            Black Swan
             (
            2010
            )
          </a>
        </li>
        <li>
          <a
            href="/movie/11"
          >
            Gravity
             (
            2013
            )
          </a>
        </li>
      </ul>
    </div>
  </section>
</div>
`;

